sdks/java/core/src/test/java/org/apache/beam/sdk/io/FileSystemsTest.java

```diff
@@ -316,6 +317,25 @@ public void testInvalidSchemaMatchNewResource() {
     assertEquals("file", FileSystems.matchNewResource("c:/tmp/f1", false));
   }
 
+  @Test
+  public void testMatchNewDirectory() {
```

Comment:
Does this test only work in Linux/MacOS? Do we need another test for Windows?

Reply:
this test works for all platforms (we have GHA for it). The initial commit was failing on windows but then fixed in followup.